### PR TITLE
MGMT-4623 Publish index image for assisted service operator bundle

### PR DESCRIPTION
**Pull request by rwsu**
to quay.io

The index is published to https://quay.io/repository/ocpmetal/assisted-service-index

Publishing the index enables users to use the latest published operator
bundle in quay.io.

Signed-off-by: Richard Su <rwsu@redhat.com>

### CODE REVIEW

**Comment by rwsu** [comment]
We need to publish the bundle image before building the index. "opm index add" pulls the bundle image.

File: Dockerfile.assisted-service-build [hunk 1]
```diff
@@ -29,5 +29,6 @@ RUN export ARCH=$(case $(arch) in x86_64) echo -n amd64 ;; aarch64) echo -n arm6
   && curl -LO ${OPERATOR_SDK_DL_URL}/operator-sdk_${OS}_${ARCH} \
   && chmod +x operator-sdk_${OS}_${ARCH} \
   && install operator-sdk_${OS}_${ARCH} /usr/local/bin/operator-sdk
+COPY --from=quay.io/operator-framework/upstream-opm-builder:v1.16.1 /bin/opm /bin
```
Comment (carbonin) [inline Dockerfile.assisted-service-build:32]: Why pull from an image? Any reason not to download from https://github.com/operator-framework/operator-registry/releases/download/v1.16.1/linux-amd64-opm?

File: Makefile [hunk 2]
```diff
@@ -430,10 +432,13 @@ operator-bundle: create-ocp-manifests
...
 operator-bundle-build:
-	podman build $(CONTAINER_BUILD_PARAMS) -f Dockerfile.bundle -t $(BUNDLE_IMAGE) .
+	docker build $(CONTAINER_BUILD_PARAMS) -f Dockerfile.bundle -t $(BUNDLE_IMAGE) .
 
 operator-bundle-update:
-	podman push $(BUNDLE_IMAGE)
+	docker push $(BUNDLE_IMAGE)
+
+operator-index-build:
+	opm index add --bundles $(BUNDLE_IMAGE) --tag $(INDEX_IMAGE) --container-tool docker
```
Comment (carbonin) [inline Makefile:444]: Why switch to docker for this?

Comment (rwsu) [reply Makefile:444]: Yuval preferred we stay on the same tooling until we switch everything over to podman.

File: Dockerfile.assisted-service-build (hunk 1 above)
Comment (rwsu) [reply Dockerfile.assisted-service-build:32]: I think it is simpler to copy from an existing image and easier to read.

**Comment by lranjbar** [comment]
Nice :+1:

**Pull request merged**